test/tests/functional/pbs_node_rampdown.py

```diff
@@ -6362,3 +6415,774 @@ def test_release_nodes_shared_server_restart_immed(self):
+        fail and schedselect will not change
+        """
+
+        jid = self.create_and_submit_job('job1_5')
+
+        self.server.expect(JOB, {'job_state': 'R',
+                                 'Resource_List.mem': '6gb',
+                                 'Resource_List.ncpus': 8,
+                                 'Resource_List.nodect': 3,
+                                 'Resource_List.select': self.job1_select,
+                                 'Resource_List.place': self.job1_place,
+                                 'schedselect': self.job1_schedselect,
+                                 'exec_host': self.job1_exec_host,
+                                 'exec_vnode': self.job1_exec_vnode}, id=jid)
+
+        cmd = [self.pbs_release_nodes_cmd, '-j', jid, self.n4]
+        ret = self.server.du.run_cmd(self.server.hostname, cmd)
+        self.assertEqual(ret['rc'], 0)
+
+        # Verify remaining job resources
+        newsel = "1:mem=2097152kb:ncpus=3+1:mem=1048576kb:ncpus=2+" + \
+                 "1:ncpus=2:mem=2097152kb"
+        new_exec_host = self.job1_exec_host
+        new_exec_vnode = self.job1_exec_vnode.replace(
+            "%s:mem=1048576kb:ncpus=1+" % (self.n4,), "")
+        self.server.expect(JOB, {'job_state': 'R',
+                                 'Resource_List.mem': '5gb',
+                                 'Resource_List.ncpus': 7,
+                                 'Resource_List.select': newsel,
+                                 'Resource_List.place': self.job1_place,
+                                 'Resource_List.nodect': 3,
+                                 'schedselect': newsel,
+                                 'exec_host': self.job1_exec_host,
+                                 'exec_vnode': new_exec_vnode}, id=jid)
+
+        # Suspend the job with qsig
+        self.server.sigjob(jid, 'suspend', runas=ROOT_USER)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid)
+
+        # Try releasing a node from suspended job
+        cmd = [self.pbs_release_nodes_cmd, '-j', jid, self.n7]
+        ret = self.server.du.run_cmd(self.server.hostname, cmd,
+                                     sudo=True)
+        self.assertNotEqual(ret['rc'], 0)
+
+        # Verify that resources won't change
+        self.server.expect(JOB, {'job_state': 'S',
+                                 'Resource_List.mem': '5gb',
+                                 'Resource_List.ncpus': 7,
+                                 'Resource_List.select': newsel,
+                                 'Resource_List.place': self.job1_place,
+                                 'Resource_List.nodect': 3,
+                                 'schedselect': newsel,
+                                 'exec_host': self.job1_exec_host,
+                                 'exec_vnode': new_exec_vnode}, id=jid)
+
+        # Release the job and make sure it is running
+        self.server.sigjob(jid, 'resume')
+        self.server.expect(JOB, {'job_state': 'R',
```

Comment:
I don't really see resuming the job as part of this test.  You are testing to see that you can't release nodes from a suspended job.  This has been done.  Resuming the job is just testing that you can resume jobs.

Reply:
resuming and also validating the job resources making sure nothing has changed. I would leave it like that.